lib/screens/app/wallet/receive.dart

```diff
@@ -382,7 +382,11 @@ class _ProductsCatalogState extends State<ProductsCatalog> {
                             color: AppColors.getColorByString(products[index].name),
                             child: Center(
                               child: Text(
-                                products[index].name.characters.first,
+                                products[index].name == null
```

Comment:
Merry Christmas :christmas_tree:

Reply:
Can be replaced to following syntax
```
products[index]?.name?.characters?.first ?? ""
```

Reply:
This is a patch  not a fix to the core issue 

We should not allow products without a name

Reply:
@7flash  Your code suggestion only checks for null, He is checking for null or empty.